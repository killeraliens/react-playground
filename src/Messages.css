.Messages {
  background-color: lightgrey;
  position: relative;
  display: block;
/*  display: flex;
  justify-content: center;
  align-items: center;*/
  padding: 20px;
  margin: 10px;

}

.messages-alert {
  background-color: red;
  color: white;
  border-radius: 50%;
  padding: 6px;
  position: absolute;
  right: -10px;
  top: -10px;
}
