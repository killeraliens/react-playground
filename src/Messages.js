import React from 'react';
// import ReactDOM from 'react-dom';
import './Messages.css';

function Messages(props) {
  const badge = props.unreadQty ? <span className="messages-alert">{props.unreadQty}</span> : null;

  return (
    <div className="Messages" >
      {badge}
      <span className="messages-name">{props.name}</span>
    </div>
  );
}

export default Messages;
