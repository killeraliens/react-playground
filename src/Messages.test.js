import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Messages from './Messages';


describe( 'Message component', () => {

    it('renders messages component', () => {
      const div = document.createElement('div');
      ReactDOM.render(<Messages />, div);
      ReactDOM.unmountComponentAtNode(div);
    })

    it('renders the UI as expected', () => {
      const tree = renderer.create(<Messages name='Messages' unreadQty={4}/>).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('does not render badge if unreadQty is zero', () => {
      const tree = renderer.create(<Messages name='Messages' unreadQty={0}/>).toJSON();
      expect(tree).toMatchSnapshot();
    });

});

