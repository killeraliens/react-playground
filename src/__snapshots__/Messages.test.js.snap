// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message component does not render badge if unreadQty is zero 1`] = `
<div
  className="Messages"
>
  <span
    className="messages-name"
  >
    Messages
  </span>
</div>
`;

exports[`Message component renders the UI as expected 1`] = `
<div
  className="Messages"
>
  <span
    className="messages-alert"
  >
    4
  </span>
  <span
    className="messages-name"
  >
    Messages
  </span>
</div>
`;

exports[`does not render badge if unreadQty is zero 1`] = `
<div
  className="Messages"
>
  <span
    className="messages-name"
  >
    Messages
  </span>
</div>
`;

exports[`renders the UI as expected 1`] = `
<div
  className="Messages"
>
  <span
    className="messages-alert"
  >
    4
  </span>
  <span
    className="messages-name"
  >
    Messages
  </span>
</div>
`;
